p2p/module_raintree_test.go

```diff
@@ -1,3 +1,5 @@
+//go:build test
```

Comment:
It's only necessary to add the test build tag to:
1. **non-test** files in a non-internal package which exports a private member for testing purposes.
2. **test files** which depend on something exported by a file which uses the build tag (assumes at least one instance of case 1). 


Are you suggesting a linter which enforces the `test` build tag on all test files or something?

I don't suppose there would really be a downside to adding the test tag in places where it's not strictly necessary. I can imagine an argument for either but am leaning towards "ubiquitous build tags" at the moment (unexpectedly):

- ubiquitous build tags
  - pros
    - consistent
    - simpler
    - better justifies necessary updates to the makefile (e.g. `-tags test`; necessary in either case)
  - cons
    - slightly unconventional
    - requires running tests via make targets OR adding tags manually (e.g. IDE run/test configs)
  - UX
    - `go test` runs no tests
- only necessary build tags
  - pros
    - minimal impact on the codebase
  - cons
    - more mental overhead
    - less obvious when test(s) didn't run because of a missing tag
  - UX
    - `go test` runs some but not all tests